app/controllers/incomeSources/add/AddIncomeSourceStartDateController.scala

```diff
@@ -195,54 +200,56 @@ class AddIncomeSourceStartDateController @Inject()(authenticate: AuthenticationP
           controllers.incomeSources.add.routes.AddIncomeSourceStartDateController.submitSoleTraderBusiness,
-          controllers.incomeSources.add.routes.AddBusinessStartDateCheckController.show(),
-          SelfEmployment.messagesPrefix
+          controllers.incomeSources.add.routes.AddIncomeSourceStartDateCheckController.showSoleTraderBusiness
         )
       case (false, UkProperty) =>
         (
           controllers.incomeSources.add.routes.AddIncomeSourceController.show(),
           controllers.incomeSources.add.routes.AddIncomeSourceStartDateController.submitUKProperty,
-          controllers.incomeSources.add.routes.CheckUKPropertyStartDateController.show(),
-          UkProperty.messagesPrefix
+          controllers.incomeSources.add.routes.AddIncomeSourceStartDateCheckController.showUKProperty
         )
       case (false, ForeignProperty) =>
         (
           controllers.incomeSources.add.routes.AddIncomeSourceController.show(),
           controllers.incomeSources.add.routes.AddIncomeSourceStartDateController.submitForeignProperty,
-          controllers.incomeSources.add.routes.ForeignPropertyStartDateCheckController.show(),
-          ForeignProperty.messagesPrefix
+          controllers.incomeSources.add.routes.AddIncomeSourceStartDateCheckController.showForeignProperty
         )
       case (true, SelfEmployment) =>
         (
           controllers.incomeSources.add.routes.AddBusinessNameController.showAgent(),
           controllers.incomeSources.add.routes.AddIncomeSourceStartDateController.submitSoleTraderBusinessAgent,
-          controllers.incomeSources.add.routes.AddBusinessStartDateCheckController.showAgent(),
-          SelfEmployment.messagesPrefix
+          controllers.incomeSources.add.routes.AddIncomeSourceStartDateCheckController.showSoleTraderBusinessAgent
         )
       case (true, UkProperty) =>
         (
           controllers.incomeSources.add.routes.AddIncomeSourceController.showAgent(),
           controllers.incomeSources.add.routes.AddIncomeSourceStartDateController.submitUKPropertyAgent,
-          controllers.incomeSources.add.routes.CheckUKPropertyStartDateController.showAgent(),
-          UkProperty.messagesPrefix
+          controllers.incomeSources.add.routes.AddIncomeSourceStartDateCheckController.showUKPropertyAgent
         )
       case (true, ForeignProperty) =>
         (
           controllers.incomeSources.add.routes.AddIncomeSourceController.showAgent(),
           controllers.incomeSources.add.routes.AddIncomeSourceStartDateController.submitForeignPropertyAgent,
-          controllers.incomeSources.add.routes.ForeignPropertyStartDateCheckController.showAgent(),
-          ForeignProperty.messagesPrefix
+          controllers.incomeSources.add.routes.AddIncomeSourceStartDateCheckController.showForeignPropertyAgent
         )
     }
   }
 
+  private def getMessagesPrefix(incomeSourceType: IncomeSourceType): String = {
```

Comment:
looks like redundant method as you can do next call: 
`
private def getMessagesPrefix(incomeSourceType: IncomeSourceType): String = incomeSourceType.addIncomeSourceStartDateMessagesPrefix
`